src/legacyFeeClaims/src/generate-merkle-tree.ts

```diff
@@ -0,0 +1,250 @@
+import { readFileSync, writeFileSync } from 'node:fs';
+import { parse } from 'csv-parse/sync';
+import { ethers } from 'ethers';
+import { MerkleTree } from 'merkletreejs';
+
+interface TokenCreatorEntry {
+  tokenAddress: string;
+  currentCreator: string;
+}
+
+interface MerkleLeaf {
+  tokenAddress: string;
+  currentCreator: string;
+  hash: string;
+}
+
+function generateLeafHash(tokenAddress: string, currentCreator: string): string {
+  // Encode the token address and current creator as a packed encoding
+  const encoded = ethers.solidityPacked(['address', 'address'], [tokenAddress, currentCreator]);
```

Comment:
This should use unpacked encoding. In Ethers, I think it's:

```suggestion
  const encoded = ethers.AbiCoder.defaultAbiCoder.encode(['address', 'address'], [tokenAddress, currentCreator]);
```